WalletWasabi.Gui/Tabs/SettingsViewModel.cs

```diff
@@ -239,73 +234,28 @@ private void Save()
-
+			config.CopyFrom(Global.Config);
 			Dispatcher.UIThread.PostLogException(async () =>
 			{
 				using (await ConfigLock.LockAsync())
 				{
 					await config.LoadFileAsync();
 
 					var network = NBitcoin.Network.GetNetwork(Network);
-					var torHost = TorHost;
-					var localNodeHost = LocalNodeHost;
-					var localNodePort = LocalNodePort;
-					var torSocks5Port = int.TryParse(TorPort, out var port) ? (int?)port : null;
-					var useTor = UseTor;
-					var somePrivacyLevel = int.TryParse(SomePrivacyLevel, out int level) ? (int?)level : null;
-					var finePrivacyLevel = int.TryParse(FinePrivacyLevel, out level) ? (int?)level : null;
-					var strongPrivacyLevel = int.TryParse(StrongPrivacyLevel, out level) ? (int?)level : null;
-					var dustThreshold = decimal.TryParse(DustThreshold, out var threshold) ? (decimal?)threshold : null;
-
-					var (configLocalHost, configLocalPort) = network == NBitcoin.Network.Main
-						? (config.MainNetBitcoinCoreHost, config.MainNetBitcoinCorePort)
-						: (network == NBitcoin.Network.TestNet
-							? (config.TestNetBitcoinCoreHost, config.TestNetBitcoinCorePort)
-							: (config.RegTestBitcoinCoreHost, config.RegTestBitcoinCorePort));
-
-					if (config.Network != network
-						|| config.TorHost != torHost
-						|| config.TorSocks5Port != torSocks5Port
-						|| config.UseTor != useTor
-						|| config.PrivacyLevelSome != somePrivacyLevel
-						|| config.PrivacyLevelFine != finePrivacyLevel
-						|| config.PrivacyLevelStrong != strongPrivacyLevel
-						|| config.DustThreshold.ToUnit(MoneyUnit.BTC) != dustThreshold
-						|| configLocalHost != localNodeHost
-						|| configLocalPort.ToString() != localNodePort
-					)
-					{
-						config.Network = network;
-						config.TorHost = torHost;
-						config.TorSocks5Port = torSocks5Port;
-						config.UseTor = useTor;
-						config.PrivacyLevelSome = somePrivacyLevel;
-						config.PrivacyLevelFine = finePrivacyLevel;
-						config.PrivacyLevelStrong = strongPrivacyLevel;
-						config.DustThreshold = Money.Coins(dustThreshold.Value);
-
-						switch (network.Name)
-						{
-							case "Main":
-								config.MainNetBitcoinCoreHost = localNodeHost;
-								config.MainNetBitcoinCorePort = int.Parse(localNodePort);
-								break;
-
-							case "TestNet":
-								config.TestNetBitcoinCoreHost = localNodeHost;
-								config.TestNetBitcoinCorePort = int.Parse(localNodePort);
-								break;
-
-							case "RegTest":
-								config.RegTestBitcoinCoreHost = localNodeHost;
-								config.RegTestBitcoinCorePort = int.Parse(localNodePort);
-								break;
-						}
-
-						await config.ToFileAsync();
-					}
+					config.Network = network;
+					config.TorHost = TorHost;
+					config.TorSocks5Port = int.TryParse(TorPort, out var port) ? port : Config.DefaultTorSock5Port;
+					config.UseTor = UseTor;
+					config.DustThreshold = decimal.TryParse(DustThreshold, out var threshold) ? Money.Coins(threshold) : Config.DefaultDustThreshold;
```

Comment:
I think the `TryParse` are not necessary here because the inputs should have been validated with the `ValidateDustThreshold`, `ValidateTorPort`, `ValidatePrivacyXXXX` first. Am I right?

Reply:
I think the validate authorize empty values. (I think it should if the user wants to go back to default for example)